### DISCUSSION TITLE
How setting up Timeline plugin in Multitrack plugin? Is it possible move timeline to seperate component and get total Duration of multitrack?

### DISCUSSION THREAD

**Discussion by yarikLev04** [katspaugh/wavesurfer.js#3408] [Q&A]
Hi, recently started working with the multitrack plugin and ran into such a problem that I don't understand how to configure the timeline as if we had a separately installed Timeline plugin. 

Also is it possible to move the timeline component to another component as shown in the image below?

<img width="938" alt="Снимок экрана 2023-12-13 в 11 21 42" src="https://github.com/katspaugh/wavesurfer.js/assets/91241935/91c5ba23-55cb-45e8-b0e4-b0e8c9cb2c19">

And another question, how realistic is it to get the total multitrack time like a Wafesurfer func getDuration?

**Comment by katspaugh** [comment]
The multitrack timeline currently doesn't have configurable options. We can think about making it parametrized.

Regarding the total duration, there's only a private maxDuration property atm. Please feel free to make a pull request to add a public getter method for it.

**Reply by yarikLev04** [reply]
Yeah it will be fine, if we can setting up it

**Reply by katspaugh** [reply]
I've added a `timelineOptions` param in multitrack 0.4.7. You can now pass a custom container.

**Comment by yarikLev04** [comment]
Got it, also @katspaugh i have another question, how we can fix this issue https://github.com/katspaugh/wavesurfer.js/issues/3395 in multitrack if I dont know any container in it

**Reply by katspaugh** [reply]
In the same way as described there, pass `splitChannels` with `height: 0` for the second channel.

**Reply by yarikLev04** [reply]
yeah, one problem for splitChannels i need to have container, but in multitrack i cant get all ids of containers for seperate waveforms or I make smth wrong?

```
const mockedStems: MultitrackTracks = [
      'https://www.mfiles.co.uk/mp3-downloads/angels-we-have-heard-on-high.mp3',
      'https://www.mfiles.co.uk/mp3-downloads/shepherds-shake-off-your-drowsy-sleep.mp3',
      'https://www.mfiles.co.uk/mp3-downloads/daisy-bell.mp3'
    ].map((url, index) => ({
      id: index,
      draggable: true,
      startPosition: 0,
      url: url,
      volume: 0.5,
      options: {
        height: 54,
        waveColor: 'rgba(255, 255, 255, 0.80)',
        progressColor: 'rgba(255, 255, 255, 0.80)',
        barWidth: 3,
        barGap: 2,
        barRadius: 1,
        barHeight: 1
      }
    }));

    const stemsWS = MultiTrack.create(mockedStems, {
      container: stemsWaveForm.current!,
      minPxPerSec: 10,
      rightButtonDrag: false,
      cursorWidth: 2,
      cursorColor: '#008EF3',
      trackBackground: 'rgba(0, 0, 0, 0.10);',
      trackBorderColor: '#494949',
      dragBounds: true
    });
```

**Reply by katspaugh** [reply]
The `container` param in `splitChannels` is actually _not_ required, the type is not entirely correct.

**Reply by yarikLev04** [reply]
oh, got it, thank you
